includes/Integrations/Ezoic.php

```diff
@@ -0,0 +1,147 @@
+<?php
+/**
+ * Class Ezoic
+ *
+ * @link      https://github.com/googleforcreators/web-stories-wp
+ *
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ */
+
+/**
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+declare(strict_types = 1);
+
+namespace Google\Web_Stories\Integrations;
+
+use Google\Web_Stories\AMP\Optimization;
+use Google\Web_Stories\AMP\Sanitization;
+use Google\Web_Stories\Context;
+use Google\Web_Stories\Exception\SanitizationException;
+use Google\Web_Stories\Service_Base;
+use Google\Web_Stories_Dependencies\AmpProject\Dom\Document;
+use Throwable;
+
+/**
+ * Class Ezoic.
+ */
+class Ezoic extends Service_Base {
+	/**
+	 * Sanitization instance.
+	 *
+	 * @var Sanitization Sanitization instance.
+	 */
+	private Sanitization $sanitization;
+
+	/**
+	 * Optimization instance.
+	 *
+	 * @var Optimization Optimization instance.
+	 */
+	private Optimization $optimization;
+
+	/**
+	 * Context instance.
+	 *
+	 * @var Context Context instance.
+	 */
+	private Context $context;
+
+	/**
+	 * Constructor.
+	 * 
+	 * @since 1.33.0
+	 *
+	 * @param Sanitization $sanitization Sanitization instance.
+	 * @param Optimization $optimization Optimization instance.
+	 * @param Context      $context Context instance.
+	 */
+	public function __construct( Sanitization $sanitization, Optimization $optimization, Context $context ) {
+		$this->sanitization = $sanitization;
+		$this->optimization = $optimization;
+		$this->context      = $context;
+	}
+
+	/**
+	 * Determines whether Ezoic is installed and active.
+	 *
+	 * @since 1.33.0
+	 *
+	 * @return bool Whether Ezoic is installed and active.
+	 */
+	protected function is_plugin_installed_active(): bool {
```

Comment:
Pro tip: implement the `Conditional` interface in this class and rename this method here to `is_needed` and make it static. Then the service will only ever be registered if Ezoic is active. That means no extra check is needed in `register()`

Reply:
Done ✅ with commit https://github.com/GoogleForCreators/web-stories-wp/pull/13354/commits/1ffbca798919dc7d2fd8d6a36e7dd65a23344397.